metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/HdfsPathDataset.java

```diff
@@ -11,14 +11,13 @@
 @ToString
 public class HdfsPathDataset extends SparkDataset {
 
-  private static String getPath(Path path, boolean includeScheme) {
+  private static String getPath(Path path, boolean includeScheme, String removePartitionPattern) {
     URI uri = path.toUri();
-
-    if (includeScheme) {
-      return uri.toString();
-    } else {
-      return uri.getHost() + uri.getPath();
+    String uriPath = includeScheme ? uri.toString() : uri.getHost() + uri.getPath();
+    if (!removePartitionPattern.equals("")) {
```

Comment:
If we returned null in previous version we can check whether its not equal to null instead of empty string.

Reply:
I changed